sdk/core/src/main/java/com/klaviyo/coresdk/config/KlaviyoConfig.kt

```diff
@@ -28,10 +28,31 @@ class MissingPermission(permission: String) : Exception("You must declare $permi
  * Stores all configuration related to the Klaviyo Android SDK.
  */
 object KlaviyoConfig : Config {
-    private const val DEBOUNCE_INTERVAL: Int = 500
-    private const val NETWORK_TIMEOUT_DEFAULT: Int = 500
-    private const val NETWORK_FLUSH_INTERVAL_DEFAULT: Int = 60000
-    private const val NETWORK_FLUSH_DEPTH_DEFAULT: Int = 20
+    /**
+     * Default value: Debounce time for fluent profile setter methods
+     */
+    private const val DEBOUNCE_INTERVAL: Int = 100
+
+    /**
+     * Default value: Network request timeout duration
+     */
+    private const val NETWORK_TIMEOUT_DEFAULT: Int = 10_000
+
+    /**
+     * Default value: Interval between flushing network queue.
+     * Also used as the basis for exponential backoff when retrying requests
+     */
+    private const val NETWORK_FLUSH_INTERVAL_DEFAULT: Int = 30_000
+
+    /**
+     * Default value: How many API requests can be enqueued before flush
+     */
+    private const val NETWORK_FLUSH_DEPTH_DEFAULT: Int = 100
+
+    /**
+     * Default value: How many retries to allow an API request before permanent failure
+     */
+    private const val NETWORK_MAX_RETRIES_DEFAULT: Int = 4
```

Comment:
Since we're just popping failed requests back on the queue, they'll only really retry every 30s. Seems maybe a little slow to me for a retry backoff since that means a failed request will be tried for 2.5mins total. 


I spose it doesn't matter since none of the requests our SDK makes are time critical. Just a thought anyway

Reply:
Ah yes I see it now, nice!